handlers/preview.go

```diff
@@ -210,6 +239,8 @@ func (f *Filter) OutputPage() http.HandlerFunc {
 				return
 			}
 
+			// Can we trust opts.TotalCount?
```

Comment:
This assumes that we are using a Dataset API version that implements pagination. We did some breaking changes in order to harness pagination to improve performance to support large datasets.
Maybe we could perform some dataset API version check during the service startup, and fail if the version is lower than the one that implements pagination? (I guess this would need to be a discussion with tech leads)